lib/ama_layout/notification.rb

```diff
@@ -0,0 +1,65 @@
+module AmaLayout
+  class Notification
+    TYPES = %i[notice warning alert].freeze
+    FORMAT_VERSION = '1.0.0'.freeze
+
+    # NOTE: The following attributes are designed to be immutable - you need
+    # make a new instance to change them. The only mutable attribute is :active.
+    attr_reader :id, :type, :header, :content, :created_at, :version
+    attr_accessor :active
+
+    def initialize(args = {})
+      args = args.with_indifferent_access
+      @id = args[:id]
+      @type = args.fetch(:type, :notice).to_sym
+      @header = args.fetch(:header)
+      @content = args.fetch(:content)
+      @created_at = parse_time(args.fetch(:created_at))
+      @version = args.fetch(:version, FORMAT_VERSION)
+      self.active = args.fetch(:active)
+      invalid_type! if TYPES.exclude?(type)
+    end
+
+    def <=>(other)
+      created_at <=> other.created_at
+    end
+
+    def active?
+      active
+    end
+
+    def dismissed?
+      !active?
+    end
+
+    def dismiss!
+      self.active = false
+      true
+    end
+
+    def digest
+      Digest::SHA256.hexdigest(type.to_s + header + content + version)
```

Comment:
Right now (this may change), the notification serialization format does not allow duplicate notifications (taking into account the above attributes).

This is by design: there are many cases where we are going to want to automatically generate notifications based on some kind of state. Let's say this generation process occurs on every login - we wouldn't want to create a ton of duplicate notifications. Instead, it would be nice if the underlying abstractions took care of that for us.

To overcome cases where we will want another of the same notification (maybe months/years later), I'm going to add a `lifespan` attribute to all notifications that will specify a duration of seconds that a notification will be visible.